TrueCar/gluestick
# Parallel builds

Ref: ##1033

How to run (npm@5 is not supported):
```
git clone https://github.com/TrueCar/gluestick
cd gluestick
npm run install:npm
cd ../
./gluestick/packages/gluestick-cli/bin/index.js new testApp --npm --dev ./gluestick
cd . testApp
./node_modules/.bin/gluestick build --vendor
./node_modules/.bin/gluestick build --client
```

If vendor DLL bundle is preset it will be used, otherwise it will create normal vendor bundle using `CommonsChunkPlugin`. Also `--skip-if-ok` option in `gluestick build --vendor` will make some checks (https://github.com/TrueCar/gluestick/blob/feature/vendor-dll/packages/gluestick/src/config/vendorDll.js#L47-L112), if every of them fail, it will conclude that vendor dll bundle is fine, and will bail from recompiling it. To build single app: `gluestick build --client -A /<appName>`. To parallely build project:
- build vendor dll `gluestick build --vendor`
- build every app using `gluestick build --client -A /<appName>`
- build server bundle `gluestick build --server`
Now you can `start -P` w/ `NODE_ENV=production` set.

`gluestick-config-legacy` has also been updated to support vendor DLL bundle, so if you're using it, you need to update it `npm i <pathToGluestickRepo>/packages/gluestick-config-legacy`

Documentation can be found [here](https://github.com/TrueCar/gluestick/blob/feature/vendor-dll/docs/Configuration.md#vendoring).

@todd should vendor DLL support also CSS files and loaders?
